test-app/tests/integration/components/input-field-test.gts

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable no-undef -- Until https://github.com/ember-cli/eslint-plugin-ember/issues/1747 is resolved... */
+/* eslint-disable simple-import-sort/imports,padding-line-between-statements,decorator-position/decorator-position -- Can't fix these manually, without --fix working in .gts */
+import { fillIn, render, setupOnerror } from '@ember/test-helpers';
+import { module, test } from 'qunit';
+
+import InputField from '@crowdstrike/ember-toucan-core/components/form/input-field';
+import { setupRenderingTest } from 'test-app/tests/helpers';
+
+
+module('Integration | Component | InputField', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    await render(<template>
+      <InputField
+        @label="Label"
+        type="text"
+        data-input
+        />
+    </template>);
+
+    const label = '[data-label]'
+    const input = '[data-input]';
+
+    assert.dom(label).exists('Expected to have label block rendered');
+    assert.dom(label).hasText('Label', 'Expected to have label text "label"');
+
+    assert.dom(input).exists('Expected to have input tag rendered');
+    assert.dom(input).hasAttribute('type', 'text');
+  });
+
+  test('it encourages @label via assert', async function (assert) {
+    assert.expect(1)
+
+    setupOnerror((e:Error) => {
+      assert.strictEqual(e.message, 'Assertion Failed: input field requires a label', 'Expected an error message if @label is not provided')
+    })
+    await render(<template>
+      {{! @glint-expect-error: should have an error here for missing @label }} 
+      <InputField
+        type="text"
+        />
+    </template>);
+
+  })
+  
+  test('it renders an error', async function (assert) {
+    await render(<template>
+      <InputField
+        @label="Label"
+        type="text"
+        @error="There is an error"
+        data-input
+        />
+    </template>);
+
+    const label = '[data-label]'
+    const input = '[data-input]';
+    const error = '[data-error]';
+   
+    const inputElement = document.querySelector('[data-input]');
+ 
+    const [id] = inputElement?.getAttribute('aria-describedby')?.trim().split(' ') ?? '';
+  
+    assert.dom(label).exists('Expected to have label component rendered');
+    assert.dom(label).hasText('Label', 'Expected to have label text "label"');
+
+    assert.dom(input).exists('Expected to have input tag rendered');
+    assert.dom(input).hasAttribute('type', 'text');
+   
+    assert.dom(error).exists('Expected to have error component rendered');
+    assert.dom(error).hasText('There is an error', 'Expected to have error text "error"');
+
+    if (id) {
+      assert.dom(error).hasAttribute('id', id);
```

Comment:
yup yup.  One thing I did for Textarea to assert this stuff: https://github.com/CrowdStrike/ember-toucan-core/pull/36/files#diff-b7acb71cb9e9389fb6732d2b7eb21e52658721f8307ab8c972d27de292a9adb3R63

```js
     let error = find('[data-error]');

     assert.dom(error).hasText('Error text');
     assert.dom(error).hasAttribute('id');

     let errorId = error?.getAttribute('id') || '';
     assert.ok(errorId, 'Expected errorId to be truthy');

     let describedby =
       find('[data-textarea]')?.getAttribute('aria-describedby') || '';
     assert.ok(
       describedby.includes(errorId),
       'Expected errorId to be included in the aria-describedby'
     );

     assert.dom('[data-textarea]').hasAttribute('aria-invalid', 'true');
```

Then no conditionals and defaulting to `|| ''` and asserting it's truthy

Reply:
Question: if both the `errorId` and `describedby` end up being `''`, is that assertion correct? Or that case won't happen because assert.ok tests for truthiness? (is that the "errorId" is truthy and `describedby` is truthy??)

Reply:
I'm still going to use that snippet btw.